apps/ios/LiveVideo/LiveVideo/Twilio/Sync/SyncStreamDocument.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  Copyright (C) 2022 Twilio, Inc.
+//
+
+import Combine
+import TwilioSyncClient
+
+class SyncStreamDocument: NSObject, SyncObjectConnecting, ObservableObject {
+    struct DocumentData {
+        struct Recording {
+            let isRecording: Bool
+            let error: String?
+        }
+        
+        let recording: Recording
+        
+        init(data: [String: Any]) throws {
+            guard
+                let recording = data["recording"] as? [String: Any],
+                let isRecording = recording["is_recording"] as? Bool
+            else {
+                throw LiveVideoError.syncObjectDecodeError
+            }
+
+            let error = recording["error"] as? String
+            
+            self.recording = Recording(isRecording: isRecording, error: error)
+        }
+    }
+        
+    @Published var isRecording = false
+    
+    /// Record errors are informative and do not end the stream, so provide a special path to handle them. It would be nice to have
+    /// a single path for all errors but that would require a significant refactor and probably make other code more complex.
+    @Published var recordError: Error?
```

Comment:
Just keeping this simple by creating a separate path for record errors because:

1. This is the first time we are pushing errors from the reference backend to clients via sync
2. The recording errors are just informative and not fatal which is different from other backend and SDK errors.
3. There is also some extra logic around recording errors so that we don't show them until after the user is connected to the stream. And so handling state for this error is more complex than other errors that are always immediately handled.

We could evolve this in the future if we have more errors like this.